client/src/Pages/ResetPasswordPage.jsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+import { toast, ToastContainer } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import { resetPassword } from "../utilities/authApi";
+import { validatePassword } from "../utilities/helperFunctions";
+
+const ResetPasswordPage = () => {
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const navigate = useNavigate();
+  const { token } = useParams();
+
+  const handlePasswordReset = async (e) => {
+    e.preventDefault();
+
+    if (password !== confirmPassword) {
+      toast.error("Passwords do not match!", {
+        position: toast.POSITION.TOP_CENTER,
+      });
+      return;
+    }
+
+    try {
+      if (!validatePassword(password)) {
+        return;
+      }
+
+      const formData = {
+        token: token,
+        password: password,
+      };
+
+      await resetPassword(formData);
+      navigate("/login");
+    } catch (error) {
+      console.log(error);
+      if (error.response.status === 404) {
+        toast.error("User with token does not exist!", {
```

Comment:
User does not know what token is.
`Password reset is not requested for this account` might be better wording. This can be same wording for 403 error code